### ISSUE TITLE
 container.Register(service, reg.Implementation, Lifestyle.Transient);

### ISSUE THREAD

**Issue by hemendpatel** [simpleinjector/SimpleInjector#567]
Is there a way to check the container data before service registration:

I am registering interface and its implementation dynamically  and getting issue if interface have multiple implementation. so wan to verify it before execution below container registration statement:

container.Register(service, reg.Implementation, Lifestyle.Transient);

**Comment by dotnetjunkie** [comment]
Can you elaborate on this? 

- Please explain what it is you want to achieve? Do you wish to append the implementation to get a list of services of the same type? Or do you wish to replace the current registration?
- If the latter (replacing), can you explain why you need to replace an existing registration and why you don't already know there already is an existing registration? How are you sure that replacing the original is the right thing? Why don't you want Simple Injector to communicate that there already is a registration? Doesn't this mean a configuration error?

**Comment by hemendpatel** [comment]
Below is my code for generic registration :

var repositoryAssembly = typeof(ApplicationService.IContactDetail).Assembly;
var registrations =
from type in repositoryAssembly.GetExportedTypes()
       where type.Namespace == "ApplicationService"
       where type.GetInterfaces().Any()
       select new { Service = type, Implementation = type };

       foreach (var reg in registrations)
       {
       	foreach (var service in reg.Service.GetInterfaces())
              {

                    IServiceProvider serviceProvider = container;

                    var obj = serviceProvider.GetService(service);
			**//I AM GETTING ISSUE IF INTERFACE HAVE MULITPLE IMPLEMENATION.
			//In multiple implementation, I am using name instance logic to get 
			// get the specific handler**
                    container.Register(service, reg.Implementation, Lifestyle.Transient);

                    
              }
       }


container.Verify();

config.DependencyResolver = new SimpleInjectorWebApiDependencyResolver(container);
config.MessageHandlers.Add(new DelegatingHandlerProxy<HelloRequestHandler>(container));

**Comment by hemendpatel** [comment]
named logic implementation:

    public interface ItesttHandlerFactory
    {
        Itest CreateNew(string name);
    }
    public class testHandlerFactory : Dictionary<string, Type>, ItesttHandlerFactory
    {
        private readonly Container container;
        public testHandlerFactory(Container container)
        {
            this.container = container;
        }
        public Itest CreateNew(string name) => (Itest)this.container.GetInstance(this[name]);
    }

**Comment by dotnetjunkie** [comment]
Ah, I see. You are auto-registering instances by reflecting over the assembly.

Let me rephrase my question: since reflecting over the types of an assembly, will produce a collection of types in a random order, and considering there is a certain interface that has multiple implementations, how do you wish that to be handled? Do you wish one implementation to be selected, and if so: which one? Or do you want to inject a collection of services of that type?

**Comment by hemendpatel** [comment]
I want to inject a collection of services if interface has multiple implementation

**Comment by dotnetjunkie** [comment]
In that case, make sure you filter out the implementations of that particular abstraction from your LINQ query, and register them separately, as follows:

``` c#
container.Collection.Register<IMyAbstraction>(repositoryAssembly);
```

**Comment by hemendpatel** [comment]
got it - thank you so much
